### ISSUE TITLE
Let TestHandler::hasRecordThatContains methods also check the context (optional)

### ISSUE THREAD

**Issue by DanielSchwiperich** [Seldaek/monolog#1002]
I would find it helpful for unit tests if the hasRecordThatContains methods not only check the message but also the context. 
This would allow to have for example unit tests not only check that the correct log message is triggered but also that the correct data is written. 

instead of 
```php
    public function hasRecordThatContains($message, $level) {...}
```

```php
    public function hasRecordThatContains($message, $level, $testContext = false) {...}
```

Making it optional would also not break anything or make other stuff suddenly pass. 

I would try to provide a PR if the idea is generally accepted.

**Comment by the-invisible-man** [comment]
I'm all for this, dealing with this myself so I can check that we're not logging items with sensitive information.

**Comment by Seldaek** [comment]
Fixed by #1159
